### PR TITLE
Refactor Log File Monitor

### PR DESCRIPTION
**Pull request by iankingori**
This PR refactors some of the Log File Monitor logic and separates some of the class specific functionality from helper functions
- moved log directory handle to utilities file
- moved create event logic and error handling to utilities file
- created an enqueue events function to centralize lock acquisition and releases plus filter out events by file name before adding them to the queue
- (fix) only read sub directory changes when include subfolders is true

### CODE REVIEW

File: LogMonitor/src/LogMonitor/FileMonitor/Utilities.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * @param bManualReset
+ * @param bInitialState
+ * 
+ * return event handle
+*/
+HANDLE CreateFileMonitorEvent(
+        _In_ BOOL bManualReset,
+        _In_ BOOL bInitialState
+) {
+    HANDLE event = CreateEvent(nullptr, bManualReset, bInitialState, nullptr);
```
Comment (bobsira) [inline LogMonitor/src/LogMonitor/FileMonitor/Utilities.cpp:20]: Their seem to be CreateEventA, CreateEventW, CreateEventExA, and CreateEventExW. Hopefully we are using the right one.

File: LogMonitor/src/LogMonitor/FileMonitor/Utilities.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+#include "pch.h"
+
+/**
+ * Warapper around Create Event API
+ * 
+ * @param bManualReset
```
Comment (profnandaa) [inline LogMonitor/src/LogMonitor/FileMonitor/Utilities.cpp:11]: I see popular C++ guides are silent on how do the function block comment but I like this one, looks clean.

**Review by profnandaa — APPROVED** [review]
LGTM.

File: LogMonitor/src/LogMonitor/FileMonitor/Utilities.cpp (hunk 1 above)
Comment (iankingori) [reply LogMonitor/src/LogMonitor/FileMonitor/Utilities.cpp:20]: I haven't changed this bit of code, I'll take a look at this in the second pass to figure out whether we will gain anything by changing the APIs

**Review by bobsira — APPROVED** [review]

**Pull request merged**